File: src/main/java/com/kaankaplan/booke/business/abstracts/PostService.java
package com.kaankaplan.booke.business.abstracts;

import com.kaankaplan.booke.core.util.results.DataResult;
import com.kaankaplan.booke.core.util.results.Result;
import com.kaankaplan.booke.dto.PostDto;
import com.kaankaplan.booke.modals.Comment;
import com.kaankaplan.booke.modals.Post;

import java.util.List;

public interface PostService {

    DataResult<Post> saveOrUpdate(PostDto postDto);

    DataResult<List<Post>> getUserFollowsPost(String userId, int pageNo, int pageSize);

    Result likePost(String userId, String postId);

    Result unlikePost(String userId, String postId);

    DataResult<Comment> addCommentToPost(String postId, Comment comment);
}

File: src/main/booke-mobile-app/services/PostService.js
import Interceptor from "../utils/interceptors/Interceptor";

export default class PostService {
    apiUrl = "post/";
    axiosInstance;

    constructor() {
        this.axiosInstance = new Interceptor().getInstance();
    }

    getUserFollowsPost(userId, pageNo, pageSize) {
        return this.axiosInstance.get(this.apiUrl + "getUserFollowsPost/" + userId + "/" + pageNo + "/" + pageSize);
    }

    addCommentToPost(postId, comment) {
        return this.axiosInstance.post(this.apiUrl + "addCommentToPost/" + postId, comment);
    }
    
    likePost(userId, postId) {
        return this.axiosInstance.post(this.apiUrl + "likePost/" + userId + "/" + postId, {});
    }

    unlikePost(userId, postId) {
        return this.axiosInstance.post(this.apiUrl + "unlikePost/" + userId + "/" + postId, {});
    }

    createPost(postDto) {
        return this.axiosInstance.post(this.apiUrl + "createPost", postDto);
    }
}